### DISCUSSION TITLE
Using a custom Drude model

### DISCUSSION THREAD

**Discussion by 10UAikawa** [NanoComp/meep#2795] [Q&A]
Hi, I was just wondering if there is a way in implement your own Drude model in MEEP? I read in the docs that if you are only interested in a narrow bandwidth, you can have a frequency permittivity: "For example, suppose you want to simulate a medium with ε=3.4+0.101i
 at a frequency 0.42 (in your Meep units), and you only care about the material in a narrow bandwidth around this frequency (i.e. you don't need to simulate the full experimental frequency-dependent permittivity). Then, in Meep, you could use meep.Medium(epsilon=3.4, D_conductivity=2*math.pi*0.42*0.101/3.4) in Python"

So, if I have a custom Drude model and I am only solving for some finite number of frequency points. Is there a way that I could use my Drude model at each and every one of those frequency points? 

For example, I could do: numpy.linspace(start_freq, stop_freq, num_freq_points) and define my function for ε. Then num_freq_points can also correspond to the number of frequency points being solved for. 

I am just not sure of how I would be able to implement this to solve for each and every frequency?

**Comment by stevengj** [comment]
You can enter a Drude model directly into Meep, no fitting required: https://meep.readthedocs.io/en/latest/Python_User_Interface/#drudesusceptibility
